encryption-keys.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"crypto/aes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+)
+
+// EncryptionKey - generic interface to encrypt/decrypt some bytes
```

Comment:
> implement it like a package.. separately to be used outside.

This looks a good idea at first glance, however we will be obliged to rename some exported functions and it will be a little hard for the developer to understand our API.

NewSymmetricKey() and NewAsymmetricKey() doesn't really reflect the functionality inside these functions but they are so simple for the developers to use.